src/storage/engines/LocalStorageEngine.ts

```diff
@@ -248,6 +249,47 @@ export class LocalStorageEngine extends StorageEngine {
     return await this.studyDatabase.getItem(participantId) as ParticipantData | null;
   }
 
+  async getParticipantTags(): Promise<string[]> {
+    if (!this._verifyStudyDatabase(this.studyDatabase)) {
+      throw new Error('Study database not initialized');
+    }
+
+    const participantData = await this.getParticipantData();
+    if (!participantData) {
+      throw new Error('Participant not initialized');
+    }
+
+    return participantData.participantTags;
+  }
+
+  async addParticipantTags(tags: string[]): Promise<void> {
+    if (!this._verifyStudyDatabase(this.studyDatabase)) {
+      throw new Error('Study database not initialized');
+    }
+
+    const participantData = await this.getParticipantData();
+    if (!participantData) {
+      throw new Error('Participant not initialized');
+    }
+
+    participantData.participantTags = [...new Set([...participantData.participantTags, ...tags])];
+    await this.studyDatabase.setItem(this.currentParticipantId as string, participantData);
+  }
+
+  async removeParticipantTags(tags: string[]): Promise<void> {
+    if (!this._verifyStudyDatabase(this.studyDatabase)) {
+      throw new Error('Study database not initialized');
+    }
+
+    const participantData = await this.getParticipantData();
+    if (!participantData) {
+      throw new Error('Participant not initialized');
+    }
+
+    participantData.participantTags = participantData.participantTags.filter((tag) => !tags.includes(tag));
+    await this.studyDatabase.setItem(this.currentParticipantId as string, participantData);
+  }
+
```

Comment:
Just a general comment for future: Is there a way to automatically wrap these functions in something that checks for participant data and study database being initialized? I'm not sure how we would go about this, but it seems tedious to have to write this over and over again. At the very least, maybe we should just create a separate function that abstracts this.